edx_notifications/server/api/consumer.py

```diff
@@ -154,10 +155,39 @@ def get(self, request):
             filters=filters,
             options=options,
         )
-
         resultset = [user_msg.get_fields() for user_msg in user_msgs]
+        response_data = {
+            'notifications': resultset,
+            'unread_count': sum([not notification['read'] for notification in resultset])
+        }
+        return Response(response_data, status.HTTP_200_OK)
+
+    def post(self, request):
+        """
+        HTTP POST Handler
+        """
+        notification_data = request.data.get('notification') or request.data.get('message')
+
+        usernames = request.data.get('usernames', [])
+        if not (notification_data and usernames):
+            return JsonResponse({"message": "Invalid notification payload"}, status=status.HTTP_400_BAD_REQUEST)
 
-        return Response(resultset, status.HTTP_200_OK)
+        user_ids = User.objects.filter(username__in=usernames).values_list('id', flat=True)
+        type_name = self.get_notification_type(notification_data['type'])
+        msg_type = get_notification_type(type_name)
+
+        msg = NotificationMessage(
+            msg_type=msg_type,
+            namespace=NAMESPACE,
+            payload=notification_data,
+        )
+        bulk_publish_notification_to_users(user_ids, msg)
+
+        return Response([], status.HTTP_200_OK)
+
+    def get_notification_type(self, notification_type):
```

Comment:
I think this would no be needed if we define appropriated notification types as suggested above.